src/operator/unary_op-inl.h

```diff
@@ -0,0 +1,139 @@
+/*!
+ * Copyright (c) 2015 by Contributors
+ * \file unary_op-inl.h
+ * \brief
+ * \author Bing Xu
+*/
+#ifndef MXNET_OPERATOR_UNARY_OP_INL_H_
+#define MXNET_OPERATOR_UNARY_OP_INL_H_
+#include <dmlc/logging.h>
+#include <dmlc/parameter.h>
+#include <mxnet/operator_util.h>
+#include <vector>
+#include "./mshadow_op.h"
+
+#if defined(__CUDACC__)
+#define XPU gpu
+#else
+#define XPU cpu
+#endif
+
+namespace mxnet {
+namespace op {
+
+template<typename xpu>
+void Transpose(const TBlob &src,
+               const EnvArguments& env,
+               TBlob *ret,
+               OpReqType req,
+               RunContext ctx) {
+  mshadow::Stream<xpu> *s = ctx.get_stream<xpu>();
+  mshadow::Tensor<xpu, 2> out = ret->FlatTo2D<xpu, real_t>(s);
+  mshadow::Tensor<xpu, 2> in = src.FlatTo2D<xpu, real_t>(s);
+  out = in.T();
+}
+
+template<typename xpu>
+void TransposeGrad(const OutputGrad& src,
+                   const EnvArguments& env,
+                   TBlob *ret,
+                   OpReqType req,
+                   RunContext ctx) {
+  mshadow::Stream<xpu> *s = ctx.get_stream<xpu>();
+  mshadow::Tensor<xpu, 2> out = ret->FlatTo2D<xpu, real_t>(s);
+  mshadow::Tensor<xpu, 2> in = src.data.FlatTo2D<xpu, real_t>(s);
+  out = in.T();
+}
+
+inline TShape TransposeShape(const TShape& shp,
+                             const EnvArguments& env) {
+  CHECK(shp.ndim() == 2)
+      << "transpose only accept two dimensional input";
+  std::vector<mshadow::index_t> ret;
+  ret.push_back(shp[1]);
+  ret.push_back(shp[0]);
+  return TShape(ret.begin(), ret.end());
+}
+
+// -------------- operator
+
+struct PadParam : public dmlc::Parameter<PadParam> {
+  TShape pad_shape;
```

Comment:
We should discuss what this parameter name should be. Ideally it should be consistent with common experience. Let us open an issue about this.